java-squid/src/main/java/org/sonar/java/se/ExplodedGraphWalker.java

```diff
@@ -0,0 +1,476 @@
+      // create
+      final SymbolicValue sv = constraintManager.createSymbolicValue(variableTree);
+      startingStates = Iterables.transform(startingStates, new Function<ProgramState, ProgramState>() {
+        @Override
+        public ProgramState apply(ProgramState input) {
+          return ProgramState.put(input, variableTree.symbol(), sv);
+        }
+      });
+
+      if (variableTree.symbol().metadata().isAnnotatedWith("javax.annotation.CheckForNull")
+        || variableTree.symbol().metadata().isAnnotatedWith("javax.annotation.Nullable")
+      ) {
+        startingStates = Iterables.concat(Iterables.transform(startingStates, new Function<ProgramState, List<ProgramState>>() {
+          @Override
+          public List<ProgramState> apply(ProgramState input) {
+            List<ProgramState> states = new ArrayList<>();
+            states.addAll(sv.setConstraint(input, NullConstraint.NULL));
+            states.addAll(sv.setConstraint(input, NullConstraint.NOT_NULL));
+            return states;
+          }
+        }));
+
+      }
+    }
+    for (ProgramState startingState : startingStates) {
+      enqueue(new ExplodedGraph.ProgramPoint(cfg.entry(), 0), startingState);
+    }
+    steps = 0;
+    while (!workList.isEmpty()) {
+      steps++;
+      if (steps > MAX_STEPS) {
+        throw new MaximumStepsReachedException("reached limit of " + MAX_STEPS + " steps for method " + tree.simpleName().name() + "in class " + tree.symbol().owner().name());
+      }
+      // LIFO:
+      node = workList.removeFirst();
+      programPosition = node.programPoint;
+      if (/* last */programPosition.block.successors().isEmpty()) {
+        // not guaranteed that last block will be reached, e.g. "label: goto label;"
+        // TODO(Godin): notify clients before continuing with another position
+        continue;
+      }
+      programState = node.programState;
+      if (programPosition.i < programPosition.block.elements().size()) {
+        // process block element
+        visit(programPosition.block.elements().get(programPosition.i), programPosition.block.terminator());
+      } else if (programPosition.block.terminator() == null) {
+        // process block exit, which is unconditional jump such as goto-statement or return-statement
+        handleBlockExit(programPosition);
+      } else if (programPosition.i == programPosition.block.elements().size()) {
+        // process block exist, which is conditional jump such as if-statement
+        checkerDispatcher.executeCheckPostStatement(programPosition.block.terminator());
+      } else {
+        // process branch
+        handleBlockExit(programPosition);
+      }
+    }
+
+    checkerDispatcher.executeCheckEndOfExecution(tree);
+    // Cleanup:
+    explodedGraph = null;
+    workList = null;
+    node = null;
+    programState = null;
+    constraintManager = null;
+  }
+
+  private void handleBlockExit(ExplodedGraph.ProgramPoint programPosition) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this method "handleBlockExit" is 16 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)